### PR TITLE
feat: Nod 575 config cache modifica funzionamento refresh

### PR DESCRIPTION
**Pull request by fparisitas**
#### List of Changes
Added configuration to manage whether to update Nodo Dei Pagamenti configuration, api-config-cache, or both via a new api exposed by api-config-cache

#### Motivation and Context
https://pagopa.atlassian.net/jira/software/c/projects/NOD/boards/410/backlog?selectedIssue=NOD-575

Aggiungere al pulsante di refresh del nodo anche il refresh della cache interna pagoPa

#### How Has This Been Tested?
by _junit_ typing `mvn clean verify`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/apiconfig/core/service/RefreshService.java [hunk 1]
```diff
@@ -10,39 +12,80 @@
...
+      if( apiConfigCacheRefresh ) {
+        callApiConfigCache();
+      }
+      if( monitoringRefresh ) {
+        response = callJobTrigger(JobTrigger.REFRESH_CONFIGURATION);
+      }
     } else {
-      response = callRefreshConfigDomain(domain);
+      if( monitoringRefresh ) {
+        response = callRefreshConfigDomain(domain);
```
[the remark this answers is not in the document]
Comment (fparisitas) [reply src/main/java/it/gov/pagopa/apiconfig/core/service/RefreshService.java:61]: ok, if the refresh is always global, I would say to remove everything where domain is used. 
Can it be okay?

File: src/main/java/it/gov/pagopa/apiconfig/core/service/RefreshService.java [hunk 2]
```diff
@@ -10,39 +12,80 @@
...
+      }
     }
-
     return response;
   }
 
+  @Async
+  private void callApiConfigCache() {
```
[the remark this answers is not in the document]
Comment (fparisitas) [reply src/main/java/it/gov/pagopa/apiconfig/core/service/RefreshService.java:68]: "I suggest you use [this MDC variables](https://github.com/pagopa/pagopa-selfcare-ms-backoffice-backend/blob/main/src/main/java/it/pagopa/selfcare/pagopa/backoffice/scheduler/CiBrokerExtractionScheduler.java#L85-L106) to populate kibana dashboard."

I thought there was a common part that already managed the tracking of calls whether they are client or server.
Looking around at the other clients in the project, I missed the fact that they put MDC when their methods were invoked.

[the remark this answers is not in the document]
Comment (jacopocarlini) [reply src/main/java/it/gov/pagopa/apiconfig/core/service/RefreshService.java:68]: you're right

**Review by jacopocarlini — COMMENTED** [review]
Great Work! 🏅

**Review by jacopocarlini — APPROVED** [review]

**Pull request merged**